### DISCUSSION TITLE
Do development and production os need to match?

### DISCUSSION THREAD

**Discussion by john3641** [tethysplatform/tethys#727] [Q&A] [answered]
If I install Tethys Platform on Ubuntu 20.04 Desktop, and develop an app, will I then be able to deploy the app to a Tethys Platform that is on a Centos server without additional complications?  Does it even matter at all, as long as I pass the app code files through GitHub?  Your response to this basic question would be much appreciated.

**Comment by swainn** [comment, marked as answer]
Hi @john3641 in theory Python is cross-platform and should work similarly on any of the OSes. However, in practice, there are some things to be aware of.

Ubuntu and CentOS are both Linux OSes, so there shouldn't be any problem developing on one and hosting in production on the other. I recommend developing apps in a Linux virtual machine or similar. That's my preference.

macOS is fairly similar to Linux from a Tethys development standpoint, so many of our developers use that for development and host on Linux without issue. 

The one to be wary of is Windows. It can definitely be done, but there are some differences in the way paths are formatted between Windows and Linux that can cause problems when you go to deploy the app. That said, I believe @shawncrawley, one of our committee members, develops apps on Windows and hosts on Linux. He could probably provide some advice if you wanted to try that route.

**Reply by john3641** [reply]
Thanks, Swainn.  This is all very useful information to me.  Once again, you've been very helpful!
